src/cockpit/channels/filesystem.py

```diff
@@ -91,19 +91,15 @@ def do_open(self, options):
 class FsReadChannel(GeneratorChannel):
     payload = 'fsread1'
 
-    def do_yield_data(self, options: Dict[str, object]) -> GeneratorChannel.DataGenerator:
-        binary = options.get('binary', False)
-        max_read_size = options.get('max_read_size')
-        # TODO: generic JSON validation
-        if max_read_size is not None and not isinstance(max_read_size, int):
-            raise ChannelError('protocol-error', message='max_read_size must be an integer')
-        if not isinstance(options['path'], str):
-            raise ChannelError('protocol-error', message='path is not a string')
+    def do_yield_data(self, options: JsonObject) -> GeneratorChannel.DataGenerator:
+        path = options.get_str('path')
+        binary = options.get_bool('binary', default=False)
```

Comment:
This is now missing the transformation from Json{Type,Key}Error into ChannelError. I suspect this is done generically in protocol.py below, please see my question there.